WcaOnRails/app/controllers/incidents_controller.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class IncidentsController < ApplicationController
+  include TagsHelper
+  before_action :set_incident, only: [:show, :edit, :update, :destroy]
+
+  # Incident should have a public summary when resolved, so not everything is
+  # WRC/Delegates-only.
+
+  before_action -> { redirect_to_root_unless_user(:can_manage_incidents?) }, except: [
+    :index,
+    :show,
+  ]
+
+  def index
+    base_model = Incident.includes(:competitions, :incident_tags)
+    if current_user&.can_view_delegate_matters?
+      @incidents = base_model.all
+    else
+      @incidents = base_model.resolved
+    end
+  end
+
+  def show
+  end
+
+  def new
+    @incident = Incident.new
+  end
+
+  def edit
+  end
+
+  def create
+    @incident = Incident.new(incident_params)
+
+    if @incident.save
+      flash[:success] = "Incident was successfully created."
+      redirect_to @incident
+    else
+      render :new
+    end
+  end
+
+  def mark_as
+    @incident = Incident.find(params[:incident_id])
+    updated_attrs = {}
+    case params[:kind]
+    when "sent"
+      updated_attrs[:digest_sent_at] = Time.now
+    when "resolved"
+      updated_attrs[:resolved_at] = Time.now
+    else
+      flash[:danger] = "Unrecognize action, expecting either 'sent' or 'resolved'."
```

Comment:
Nit: unrecognize -> unrecognized. Also, I think it's more futureproof to do something like "Unrecognized action: '#{params[:kind]}'". If not, when a new kind is added, it's likely people will forget to update this message.